src/internal/components/tab-trap/index.tsx

```diff
@@ -18,4 +18,44 @@ const TabTrap = ({ focusNextCallback, disabled = false }: TabTrapProps) => {
   return <div tabIndex={disabled ? -1 : 0} onFocus={focusNextCallback} />;
 };
 
+export const TabTrapBefore = ({
+  containerRef,
+  disabled = false,
+}: {
+  containerRef: React.RefObject<HTMLElement>;
+  disabled?: boolean;
+}) => {
+  const focusLastElement = () =>
+    containerRef.current && (getLastFocusableElement(containerRef.current) as HTMLElement)?.focus();
+
+  return <TabTrap disabled={disabled} focusNextCallback={focusLastElement} />;
+};
+
+export const TabTrapAfter = ({
+  containerRef,
+  disabled = false,
+}: {
+  containerRef: React.RefObject<HTMLElement>;
+  disabled?: boolean;
+}) => {
+  const focusLastElement = () =>
+    containerRef.current && (getFirstFocusableElement(containerRef.current) as HTMLElement)?.focus();
+
+  return <TabTrap disabled={disabled} focusNextCallback={focusLastElement} />;
+};
+
+function getFirstFocusableElement(container: HTMLElement) {
+  const focusableElements = getFocusableElements(container);
+  return focusableElements[0];
+}
+
+function getLastFocusableElement(container: HTMLElement) {
+  const focusableElements = getFocusableElements(container);
+  return focusableElements[focusableElements.length - 1];
+}
+
+function getFocusableElements(container: HTMLElement) {
+  return container.querySelectorAll('button, input, select, textarea, [href], [tabindex]:not([tabindex="-1"])');
```

Comment:
That looks great, I am happy to adopt it as part of this PR 👍

Reply:
So I didn't fully understand the focus-lock implementation and instead I've taken a simplified version.

I think it would make sense to add the complete one separately with test coverage on those corner cases that you mentioned. We can also consider adding it to the toolkit, wdyt?

Reply:
It is not very great to have semi-working implementation (because if not now, where else do we have time to finish this), but I believe you know what you are doing with this functionality. Approved

Reply:
There are more places that can benefit from the new focus-lock approach. I will create a BRF task to migrate those to the new implementation with adding the necessary test coverage.